src/test/java/com/contentgrid/gateway/runtime/session/PartitionedWebSessionManagerTest.java

```diff
@@ -0,0 +1,141 @@
+
+        @GetMapping("/test")
+        @ResponseBody
+        String test(WebSession session) {
+            // explicitly start session to trigger session cookie
+            session.start();
+
+            return "OK";
+        }
+    }
+
+    @Autowired
+    ApplicationContext context;
+
+    private WebTestClient client;
+
+    @BeforeEach
+    public void setup() {
+        this.client = WebTestClient
+                .bindToApplicationContext(this.context)
+                .apply(springSecurity())
+                .configureClient()
+                .filter(new LoggingExchangeFilterFunction(log::info))
+                .responseTimeout(Duration.ofHours(1))
+                .build();
+    }
+
+    @Test
+    void sessionCookie_attributes() {
+
+        var sessionCookie = client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentegrid.app/test")
+                .exchange()
+                .expectStatus().isOk()
+                .expectCookie().exists(SESSION_COOKIE_NAME)
+                .returnResult(Void.class).getResponseCookies().getFirst(SESSION_COOKIE_NAME);
+
+        assertThat(sessionCookie).isNotNull();
+        assertThat(sessionCookie.getValue()).isNotEmpty();
+
+        // HttpOnly: session cookie should not be accessible to javascript
+        assertThat(sessionCookie.isHttpOnly()).isTrue();
+        // Secure: only send session cookie over HTTPS
+        assertThat(sessionCookie.isSecure()).isTrue();
+        // Max-Age: Negative Max-Age means cookie is deleted when the browser session ends
+        assertThat(sessionCookie.getMaxAge()).isNegative();
+        // Domain: specifying "Domain" is less restrictive, because that also implies subdomains
+        assertThat(sessionCookie.getDomain()).isNull();
+        // SameSite: Lax; cookie is sent in a first-party context or following a link to the origin
+        assertThat(sessionCookie.getSameSite()).isEqualTo("Lax");
+    }
+
+    @Test
+    void sessionCookie_resumingSession() {
+
+        var sessionCookie = client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentegrid.app/test")
+                .exchange()
+                .expectStatus().isOk()
+                .expectCookie().exists(SESSION_COOKIE_NAME)
+                .returnResult(Void.class).getResponseCookies().getFirst(SESSION_COOKIE_NAME);
+
+        assertThat(sessionCookie).isNotNull();
+
+        // request including the session cookie to the same domain
+        // should NOT rotate session id
+        client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentegrid.app/test")
+                .cookie("SESSION", sessionCookie.getValue())
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().doesNotExist("Set-Cookie");
```

Comment:
Existence/non-existence of a Set-Cookie header doesn't really verify a lot, does it?

I think a better test would be to have `/test` store/return some attribute in the session.

Like, the first time, if nothing is present in the session, generate a random value and store it in some attribute.
Afterwards, when the attribute is present in the session, you return that value from the session.

That conclusively checks that sessions for different domains are different, and that the same session is resumed for the same domain.

Reply:
Addressed this by:
1. reading the `WebTestClient-Request-Id` from the request header
2. storing it in the session
3. return the previous value in the body